src/modules/settings/BlockedSkills.svelte

```diff
@@ -0,0 +1,226 @@
+<script>
+import getValueJSON from '../system/getValueJSON';
+import setValue from '../system/setValue';
+import {
+  checkSkill,
+  clearCheckedSkills,
+  getCheckedSkills,
+  submitSkillChanges,
+} from './blockedSkills';
+
+let blockedSkillLists = getValueJSON('blockedSkillLists');
+let newListName = '';
+let error = false;
+let listName = '';
+
+function checkForDuplicates(list) {
+  return blockedSkillLists.some((l) => {
+    if (l.name === list.name) {
+      return false;
+    }
+    if (l.skills.length !== list.skills.length) {
+      return false;
+    }
+    return l.skills.every((s, i) => s === list.skills[i]);
+  });
+}
+
+const loadListChecks = [
+  [
+    (list) => !list,
+    (list) => `Unable to find list "${list.name}". Try refreshing the page.`,
+  ],
+];
+
+const updateListChecks = [
+  [
+    (list) => !blockedSkillLists.some((l) => l.name === list.name),
+    (list) => `Cannot find list named "${list.name}". Try refreshing the page.`,
+  ],
+  [
+    (list) => checkForDuplicates(list),
+    () => 'You already have a list with these skills.',
+  ],
+];
+
+const newListChecks = [
+  [
+    (list) => !list.name.length,
+    () => 'You must name your list.',
+  ],
+  [
+    (list) => list.name.length > 50,
+    () => 'List names can only be at most 50 characters long.',
+  ],
+  [
+    (list) => blockedSkillLists.some((l) => l.name === list.name),
+    (list) => `You already have a list named ${list.name}.`,
+  ],
+  [
+    (list) => checkForDuplicates(list),
+    () => 'You already have a list with these skills.',
+  ],
+];
+
+function checkForErrors(checks, ...args) {
```

Comment:
already pure function, can be moved to import

Reply:
`checkForErrors` isn't pure though. It updates the error text.

I'd need to write an additional function to link the two. What would be best?

Reply:
ah good point, i totally forgot about that.
there are a couple of ways to do this. we could use a store but i want to keep it simple.
it's bizarre how something seemingly straightforward gets bogged down when you get into the nitty gritty details!